languages/python/reference/exercise-concepts/binary-search-tree.md

```diff
@@ -0,0 +1,21 @@
+# Concepts of binary-search-tree
+
+[Example implementation](https://github.com/exercism/python/blob/master/exercises/binary-search-tree/example.py)
+
+
+## General
+- class: a general comprehension of class and and how it works is required
+- class members: user must know how members of a class work
+- class methods: user must know how methods of a class work inside and outside the class
+- Implied Argument: user needs to know how to use argoument `self` in a class
+- Dunder Methods: user needs to know when to use dunder methods `__init__` and `__str__`
+- Constructor: user needs to know how to build an object using its constructor
+- None: user needs to know the meaning of assign `None` to a variable
+- Boolean: concept required to solve the exercise
+- Loops: concept required to solve the exercise
+- Integer comparison: concept required to solve the exercise
+- Recursion: the use of recursion could be useful for the resolution of this exercise
```

Comment:
This might be understating the importance of recursion in a recursive data structure. ;-)